### ISSUE TITLE
E1101: Class 'value' has no 'any' member (no-member) for psycopg connection class

### ISSUE THREAD

**Issue by sashko1988** [pylint-dev/pylint#10222]
### Bug description

```python
Pylint can't resolve psycopg connection class members, if connection is assigned to variable before using its context manager.

that fails:
`"""dummy docstring"""
import psycopg

client = psycopg.connect("dbname=test user=postgres")
with client:
    version = client.execute(
        """
        CREATE TABLE test (
            id serial PRIMARY KEY,
            num integer,
            data text)
        """
    ).fetchone()

`

while the following passes:
`"""dummy docstring"""
import psycopg


with psycopg.connect("dbname=test user=postgres") as client:
    version = client.execute(
        """
        CREATE TABLE test (
            id serial PRIMARY KEY,
            num integer,
            data text)
        """
    ).fetchone()

`
```

### Configuration

```ini

```

### Command used

```shell
pylint src/strange_stuff/failing_module.py
```

### Pylint output

```python
************* Module strange_stuff.failing_module
src/strange_stuff/failing_module.py:6:14: E1101: Class 'value' has no 'execute' member (no-member)
```

### Expected behavior

No error triggered

### Pylint version

```shell
pylint 3.3.4
astroid 3.3.8
Python 3.12.8 (main, Dec 10 2024, 14:53:19) [Clang 16.0.0 (clang-1600.0.26.4)]
```

### OS / Environment

macos 15.3, ubuntu 22.04

### Additional dependencies

```python
astroid==3.3.8
dill==0.3.9
isort==6.0.0
mccabe==0.7.0
platformdirs==4.3.6
psycopg==3.2.4
pyenchant==3.2.2
pylint==3.3.4
tomlkit==0.13.2
typing_extensions==4.12.2
```

**Comment by jtkmckenna** [comment]
Came here to reference my issue on psycopg (and how it got listed as a pylint issue there) https://github.com/psycopg/psycopg/issues/1003

My example was similar:
```python
"""Module docstring."""
import psycopg

def get_cursor():
    """Function docstring."""
    conn = psycopg.connect("dbname=news")
    return conn.cursor()
```
Getting a warning:
```
pip install psycopg==3.2.4
pylint my_test_file.py

************* Modulemy_test_file
bad_psycopg.py:7:11: E1101: Class 'value' has no 'cursor' member (no-member)
------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)

```

**Comment by sashko1988** [comment]
@jtkmckenna, nice investigation.
